contracts/modules/v2/StakingPool.sol

```diff
@@ -507,6 +507,131 @@ contract StakingPool is IStakingPool, ERC721 {
     }
   }
 
+  /// Extends the period of an existing deposit until a tranche that ends further into the future
+  ///
+  /// @dev Only the NFT owner and the authorized addresses can call this function. Pool manager
+  /// NFTs cannot be extended.
+  ///
+  /// @param tokenId           The id of the NFT that proves the ownership of the deposit.
+  /// @param initialTrancheId  The id of the tranche the deposit is already a part of.
+  /// @param newTrancheId      The id of the new tranche determining the new deposit period.
+  /// @param topUpAmount       An optional amount if the user wants to also increase the deposit
+  ///                          amount.
+  function extendDeposit(
+    uint tokenId,
+    uint initialTrancheId,
+    uint newTrancheId,
+    uint topUpAmount
+  ) public {
+    uint _firstActiveTrancheId = block.timestamp / TRANCHE_DURATION;
+    uint maxTranche = _firstActiveTrancheId + MAX_ACTIVE_TRANCHES;
+    require(tokenId != 0, "StakingPool: Invalid token id");
```

Comment:
a bit unclear why != 0. Is the NFT 0 the special NFT unextendable?

could use a comment